.github/workflows/functions_cd.yml

```diff
@@ -0,0 +1,138 @@
+name: Continuous Delivery on prod functions
```

Comment:
From the name, one can deduce that this action deploys the function while applying infrastructure changes. Better name: "Appy Terraform changes for domain functions" or something alike.